server/reload.go

```diff
@@ -115,41 +233,86 @@ func (s *Server) diffOptions(newOpts *Options) ([]option, error) {
 		}
 		switch strings.ToLower(field.Name) {
```

Comment:
at some point, we may want to refactor our config handling to allow config sections to "Register" for being able to be reloaded. For right now, I think any sort of Command or Strategy pattern on reload alone would be roughly equivalent to this switch statement.